drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c

```diff
@@ -0,0 +1,488 @@
+
+	/* tx direction has memory as source and periph as dest. */
+	blk_cfg->source_address = (uint32_t)framebuf;
+
+	/* Destination is FLEXIO Shifters */
+	blk_cfg->dest_address = FLEXIO_MCULCD_GetTxDataRegisterAddress(flexio_lcd);
+	blk_cfg->block_size = desc->buf_size;
+	/* Transfer in each DMA loop is based on the number of shifters used */
+	stream->dma_cfg.source_burst_length = num_of_shifters * 4;
+
+	stream->dma_cfg.head_block = &stream->dma_blk_cfg;
+	/* Give the client dev as arg, as the callback comes from the dma */
+	stream->dma_cfg.user_data = (struct device *)dev;
+
+	/* Configure the DMA */
+	dma_config(lcdif_data->dma_tx.dma_dev, lcdif_data->dma_tx.channel, &stream->dma_cfg);
+
+	/* The DMA driver does not support setting this Modulo value which is required
+	 * in case of the flexio module to form a circular chain between the Shift buffer
+	 * in the FLEXIO module.
+	 * TODO: Look to move this into the Zephyr EDMA HAL
+	 */
+	EDMA_SetModulo(DMA0, lcdif_data->dma_tx.channel, kEDMA_ModuloDisable,
+		       flexio_lcdif_get_edma_modulo(num_of_shifters));
+
+	/* For 6800, de-assert the RDWR pin. */
+	if (kFLEXIO_MCULCD_6800 == flexio_lcd->busType) {
+		flexio_lcdif_set_rd_wr(false, (void *)dev);
+	}
+
+	nxp_flexio_lock(config->flexio_dev);
+	FLEXIO_MCULCD_SetMultiBeatsWriteConfig(flexio_lcd);
+	FLEXIO_MCULCD_EnableTxDMA(flexio_lcd, true);
+	nxp_flexio_unlock(config->flexio_dev);
+
+	/* Start the data transfer */
+	dma_start(lcdif_data->dma_tx.dma_dev, lcdif_data->dma_tx.channel);
+
+	/* Wait for transfer done. */
+	k_sem_take(&lcdif_data->transfer_done, K_FOREVER);
+
+	return 0;
+}
+
+static int mipi_dbi_flexio_lcdif_command_write(const struct device *dev,
+					       const struct mipi_dbi_config *dbi_config,
+					       uint8_t cmd, const uint8_t *data_buf,
+					       size_t len)
+{
+	const struct mcux_flexio_lcdif_config *config = dev->config;
+	FLEXIO_MCULCD_Type *flexio_lcd = config->flexio_lcd_dev;
+	int ret;
+
+	ARG_UNUSED(dbi_config);
+
+	ret = mipi_dbi_flexio_lcdif_configure(dev, dbi_config);
+	if (ret) {
+		return ret;
+	}
+
+	FLEXIO_MCULCD_StartTransfer(flexio_lcd);
+
+	nxp_flexio_lock(config->flexio_dev);
+
+	FLEXIO_MCULCD_WriteCommandBlocking(flexio_lcd, cmd);
+
+	if ((data_buf != NULL) && (len != 0)) {
+		flexio_lcdif_write_data_array(flexio_lcd, data_buf, len);
+	}
+
+	nxp_flexio_unlock(config->flexio_dev);
+
+	FLEXIO_MCULCD_StopTransfer(flexio_lcd);
+
+	return kStatus_Success;
+
+}
+
+static int mipi_dbi_flexio_lcdif_reset(const struct device *dev, uint32_t delay)
+{
+	int err;
+	const struct mcux_flexio_lcdif_config *config = dev->config;
+
+	/* Check if a reset port is provided to reset the LCD controller */
+	if (config->reset.port == NULL) {
+		return 0;
+	}
+
+	/* Reset the LCD controller. */
+	err = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_HIGH);
+	if (err) {
+		return err;
+	}
+
+	err = gpio_pin_set_dt(&config->reset, 0);
+	if (err < 0) {
+		return err;
+	}
+
+	k_usleep(delay);
```

Comment:
By the API definition, this delay should be in milliseconds (`k_msleep`) not microseconds (`k_usleep`)